File: Solution/2341.数组能形成多少数对/2341.数组能形成多少数对.java
/*
 * @Descroption: LeetCode 2341. 数组能形成多少数对
 * @Author: EmoryHuang
 * @Date: 2023-02-16 09:20:55
 * @Method:
 * 简单题，哈希表记录数字出现次数
 */

class Solution {
    public int[] numberOfPairs(int[] nums) {
        Map<Integer, Boolean> cnt = new HashMap<>();
        int ans = 0;
        for (int num : nums) {
            cnt.put(num, !cnt.getOrDefault(num, false));
            if (!cnt.get(num)) {
                ans++;
            }
        }
        return new int[] { ans, nums.length - ans * 2 };
    }
}

File: Solution/2341.数组能形成多少数对/2341.数组能形成多少数对.py
'''
Descroption: LeetCode 2341. 数组能形成多少数对
Author: EmoryHuang
Date: 2023-02-16 09:15:30
Method:
简单题，哈希表记录数字出现次数
'''


class Solution:
    def numberOfPairs(self, nums: List[int]) -> List[int]:
        cnt = Counter(nums)
        ans = [0] * 2
        for k, v in cnt.items():
            ans[0] += v // 2
            ans[1] += v % 2
        return ans
